.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "main" ]
+    tags:
+      - "*.*.*"
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    # https://docs.docker.com/build/ci/github-actions/manage-tags-labels/
+    - name: Docker meta
+      id: meta
+      uses: docker/metadata-action@v5
+      with:
+        # list of Docker images to use as base name for tags
+        images: |
+          hackinsdn/mininet-sec
+        # generate Docker tags based on the following events/attributes
+        tags: |
+          type=ref,event=branch
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha
```

Comment:
Please double check the point we discussed about the space consuming on the docker hub because of having multiple tags

Reply:
According to documentation [0] and public oficial forum [1] there is no limit in storage size and tags for docker images. So i think it's all good for now :)

[0]: https://docs.docker.com/docker-hub/usage/
[1]: https://forums.docker.com/t/does-docker-hub-have-a-size-limitation-on-repos-or-images/10154